### PR TITLE
Add contributors/devel/staging.md

### PR DESCRIPTION
**Pull request by sttts**
Document the publishing bot.

### CODE REVIEW

File: contributors/devel/staging.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Staging Directory and Publishing
+
+The [staging/ directory](https://github.com/kubernetes/kubernetes/tree/master/staging) of Kubernetes contains a number of pseudo repositories ("staging repos"). They are symlinked into Kubernetes' [vendor/ directory](https://github.com/kubernetes/kubernetes/tree/master/vendor/k8s.io) for Golang to pick them up.
```
Comment (cblecker) [inline contributors/devel/staging.md]: Let's use git.k8s.io urls:
- https://git.k8s.io/kubernetes/staging
- https://git.k8s.io/kubernetes/vendor/k8s.io

File: contributors/devel/staging.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+We publish the staging repos using the [publishing bot](https://github.com/kubernetes/publishing-bot). It uses `git filter-branch` essentially to [cut the staging directories into separate git trees](https://de.slideshare.net/sttts/cutting-the-kubernetes-monorepo-in-pieces-never-learnt-more-about-git) and pushing the new commits to the corresponding real repositories in the [kubernetes organization on Github](https://github.com/kubernetes).
```
Comment (cblecker) [inline contributors/devel/staging.md]: - https://git.k8s.io/publishing-bot

File: contributors/devel/staging.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+The list of staging repositories and their published branches are listed in [publisher.go inside of the bot](https://github.com/kubernetes/publishing-bot/blob/master/cmd/publishing-bot/publisher.go). Though it is planned to move this out into the k8s.io/kubernetes repository.
```
Comment (cblecker) [inline contributors/devel/staging.md]: - https://git.k8s.io/publishing-bot/cmd/publishing-bot/publisher.go

File: contributors/devel/staging.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+- kube-aggregator
+- metrics
+- sample-apiserver
+- sample-controller
+
+Kubernetes tags (e.g., v1.9.1-beta1) are also applied automatically to the published repositories, prefixed with kubernetes- (e.g., kubernetes-1.9.1-beta1). The client-go semver tags (on client-go only!) including release-notes are still done manually.
+
+The semver tags are still the (well tested) official releases. The kubernetes-1.x.y tags have limited test coverage (we have some automatic tests in place in the bot), but can be used by early adopters of client-go and the other libraries. Moreover, they help to vendor the correct version of k8s.io/api and k8s.io/apimachinery.
+
+If further repos under staging are need, adding them to the bot is easy. Contact one of the [owners of the bot](https://github.com/kubernetes/publishing-bot/blob/master/OWNERS).
```
Comment (cblecker) [inline contributors/devel/staging.md]: - https://git.k8s.io/publishing-bot/OWNERS

File: contributors/devel/staging.md (hunk 1 above)
Comment (sttts) [reply contributors/devel/staging.md]: Why don't we have these redirects without the `git.` prefix? Would make go package map to github pages.

**Comment by sttts** [comment]
@cblecker updated the links.

**Comment by sttts** [comment]
> /test all [submit-queue is verifying that this PR is safe to merge]

We really have a submit queue here? :)

**Pull request merged**